### PR TITLE
Fix pokemon forms appearing twice in search

### PR DESCRIPTION
**Pull request by Teremiare**

### CODE REVIEW

File: js/search.js [hunk 1]
```diff
@@ -155,7 +155,7 @@
 		var queryAlias;
 		if (query in BattleAliases) {
 			queryAlias = toId(BattleAliases[query]);
-			if (query === 'pika' || query === 'bee' || query === 'sab') searchPasses.shift();
+			if (queryAlias.search(query) === 0) searchPasses.shift();
```
Comment (Zarel) [inline js/search.js]: ...no

Comment (Zarel) [inline js/search.js]: This removes `zard` as being an alias of `charizard`, which is bad.

Comment (Zarel) [inline js/search.js]: Oh, wait, `=== 0`

Comment (Zarel) [inline js/search.js]: Why not slice and compare? In theory that should be faster than `.search`

Comment (Zarel) [inline js/search.js]: What's the difference between `.search` and `.indexOf`?

Comment (Zarel) [inline js/search.js]: Also, `searchPasses.shift()` removes the non-alias search. This makes it impossible to find "Sablenite" on `sab`. We should probably do it the other way around.

Comment (Teremiare) [inline js/search.js]: I have tested it and it worked for every alias as far as I could see. Also I guess .indexOf would be better here, gonna change that.

**Comment by Teremiare** [comment]
Ah yeah, the item thing is a problem

File: js/search.js (hunk 1 above)
Comment (Teremiare) [inline js/search.js]: Ah yeah, the item thing is a problem

**Comment by Teremiare** [comment]
Now it only shifts on pokemon. Also changed it to indexOf as there was no need to use search there.

File: js/search.js (hunk 1 above)
Comment (Zarel) [inline js/search.js]: Yes, it is.

File: js/search.js [hunk 2]
```diff
@@ -155,7 +155,7 @@
 		var queryAlias;
 		if (query in BattleAliases) {
 			queryAlias = toId(BattleAliases[query]);
-			if (query === 'pika' || query === 'bee' || query === 'sab') searchPasses.shift();
+			if (queryAlias.indexOf(query) === 0 && qType === 'pokemon') searchPasses.shift();
```
Comment (Zarel) [inline js/search.js]: Didn't I say slice and compare would be better than either search or indexOf? :|

Comment (Zarel) [inline js/search.js]: Also, are you sure this doesn't lead to any other conflicts?

Comment (Zarel) [inline js/search.js]: You fixed the item conflict but not the other ones... might be easier to skip the alias rather than skipping the original query.

**Comment by Zarel** [comment]
This still isn't skipping the alias...

**Comment by Teremiare** [comment]
Now it just ignores the alias if the name starts with it instead of skipping the original query

**Pull request merged**